docs/commit-status-controllers/development-best-practices.md

```diff
@@ -0,0 +1,124 @@
+# Commit Status Controller Best Practices
+
+This document outlines best practices for implementing custom commit status controllers in GitOps Promoter.
```

Comment:
Maybe also link the [youtube video](https://www.youtube.com/watch?v=Usi38ly1pe0)?